app/models/reaction.rb

```diff
@@ -77,8 +77,16 @@ def cached_any_reactions_for?(reactable, user, category)
     #
     # @return [TrueClass] yup, they're spamming the system.
     # @return [FalseClass] they're not (yet) spamming the system
-    def user_has_been_given_too_many_spammy_reactions?(user:)
-      article_vomits.where(reactable_type: "Article", reactable_id: user.articles.pluck(:id)).size > 2
+    def user_has_been_given_too_many_spammy_article_reactions?(user:, threshold: 2)
+      article_vomits.where(reactable_type: "Article", reactable_id: user.articles.pluck(:id)).size > threshold
```

Comment:
Rails has a short-hand for `pluck(:id)` called `ids`. Both generate the same query, `SELECT "users"."id" FROM "users"`.

```suggestion
      article_vomits.where(reactable_type: "Article", reactable_id: user.ids).size > threshold
```